### PR TITLE
Fix for #38

### LINKED ISSUE #38: Execute search on ArrowDown if no suggestions provided?
## **I'm submitting a ...**
  - [ ] bug report
  - [X] feature request
  - [ ] support request

## **What is the current behavior?**

Currently, if there's no suggestions (suggestion list is empty), the list will be shown only after pressing Enter (executing `list/getSuggestions` function). The intuitive `arrow down` (show-list-key) will do nothing in this context.

## **What is the expected behavior?**

When there're no suggestions, the pressing of `arrow down` (show-list-key) would also work the same way as pressing the `Enter` (search-key), but only in this specific context. In all other contexts, the `arrow down` (show-list-key) would only show the list of suggestions.

Such a behaviour currently would only be achieved only by including the `arrow down` (show-list-key) in same control section where  the `Enter` (search-key) is now. Which is not convenient at all.

## **What is the motivation / use case for changing the behavior?**

In the case of showing the suggestions when there's none, it seems convenient to me to search for them.

### PR DESCRIPTION
**Pull request by Raiondesu**
## **Please check if the PR fulfills these requirements**
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?**

Feature fix. X)
This fixes #38 

## **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

This PR also proposes method name changes as deprecations.
Nothing serious.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -407,13 +407,13 @@ You can use these to imitate some of the component's behaviours.
 
 | Name | Arguments | Description |
 |------|-----------|-------------|
-|`onInputClick`| HTML click event | Fires whenever the input is being clicked. |
+|`showSuggestions`| HTML click event | Fires whenever the input is being clicked. Shows suggestion list (gets them from the source if none is present). Used to be named `onInputClick`. |
```
Comment (kaskar2008) [inline README.md]: I think it is better to keep the old name and set a lable `will be renamed` or something. And set `showSuggestions` as an alias

File: example/src/README.html [hunk 2]
```diff
@@ -535,9 +535,9 @@ <h4 id="ref-event-handlers"><a class="header-link" href="#ref-event-handlers"></
 </thead>
 <tbody>
 <tr>
-<td><code>onInputClick</code></td>
+<td><code>showSuggestions</code></td>
```
Comment (kaskar2008) [inline example/src/README.html]: Set the `alias` label and keep the old name

File: example/src/README.html [hunk 3]
```diff
@@ -565,9 +565,9 @@ <h4 id="ref-event-handlers"><a class="header-link" href="#ref-event-handlers"></
 <td>Fires on component keyup. Internally used for hiding the list.</td>
 </tr>
 <tr>
-<td><code>onArrowKeyDown</code></td>
+<td><code>moveSelection</code></td>
```
Comment (kaskar2008) [inline example/src/README.html]: Same as `showSuggestions`

**Comment by Raiondesu** [comment]
Changes have been applied, @kaskar2008-sensei.

**Pull request merged**